src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java

```diff
@@ -110,6 +125,18 @@ public void auditStarted(AuditEvent event) {
 
     @Override
     public void auditFinished(AuditEvent event) {
+        fileMessages.forEach((fileName, messages) -> {
+            writeFileOpeningTag(fileName);
```

Comment:
Don't we need to call writeFileClosingTag in this method?
Or message without file closing tag should mean, that something went wrong during file processing?